File: src/com/jikexueyuan/Course05_5/_058LengthOfLastWord.c
#include <stdio.h>
#include <string.h>
	int lengthOfLastWord(char* s) {
	       if(s==NULL||strlen(s)==0){
	    	   return 0;
	       }else{
	    	   int count=0;
	    	   int i=strlen(s)-1;
	    	   for(;i>=0;i--){
	    		   if(s[i]==' '){

	    		   }else{
	    			   break;
	    		   }
	    	   }
	    	   for(;i>=0;i--){
	    		   if(s[i]==' '){
	    			   break;
	    		   }else{
	    			   count++;
	    		   }
	    	   }
	    	   return count;
	       }
	    }
	int main(int argc, char **argv) {
		char* str="   Thank you very much    ";
		printf("%d\n",lengthOfLastWord(str));
		return 0;
	}

File: src/com/jikexueyuan/Course05_5/_058LengthOfLastWord.java
package com.jikexueyuan.Course05_5;

import org.junit.Test;

/**
 * æœ?Žä¸?¸ªå•è¯çš„é•¿åº?
 *
 */
public class _058LengthOfLastWord {
	  public int lengthOfLastWord(String s) {
	       if(s==null||s.length()==0){
	    	   return 0;
	       }else{
	    	   int count=0;
	    	   int i=s.length()-1;
	    	   for(;i>=0;i--){
	    		   if(s.charAt(i)==' '){
	    			   
	    		   }else{
	    			   break;
	    		   }
	    	   }
	    	   for(;i>=0;i--){
	    		   if(s.charAt(i)==' '){
	    			   break;
	    		   }else{
	    			   count++;
	    		   }
	    	   }
	    	   return count;
	       }
	    }
	  @Test
	  public void test(){
		  String str="   Thank you very much    ";
		  System.out.println(lengthOfLastWord(str));
	  }
}
